### ISSUE TITLE
test_rootoftools.py fails on i686

### ISSUE THREAD

**Issue by deep-42-thought** [sympy/sympy#13234]
Hi,

while trying to compile python2-sympy-1.1.1 for archlinux32 (the i686 continuitation fork of archlinux), the following test failed:

```
________________________________________________________________________________
___________ sympy/polys/tests/test_rootoftools.py:test_CRootOf_evalf ___________
  File "/build/python-sympy/src/sympy-sympy-1.1.1-py2/sympy/polys/tests/test_rootoftools.py", line 225, in test_CRootOf_evalf
    a, b = rootof(eq, 1).n(2).as_real_imag()
  File "sympy/core/evalf.py", line 1394, in evalf
    result = evalf(self, prec + 4, options)
  File "sympy/core/evalf.py", line 1292, in evalf
    xe = x._eval_evalf(prec)
  File "sympy/polys/rootoftools.py", line 644, in _eval_evalf
    x0 = mpc(*map(str, interval.center))
  File "/usr/lib/python2.7/site-packages/mpmath/ctx_mp_python.py", line 374, in __new__
    imag = cls.context.mpf(imag)
  File "/usr/lib/python2.7/site-packages/mpmath/ctx_mp_python.py", line 77, in __new__
    v._mpf_ = mpf_pos(cls.mpf_convert_arg(val, prec, rounding), prec, rounding)
  File "/usr/lib/python2.7/site-packages/mpmath/ctx_mp_python.py", line 84, in mpf_convert_arg
    if isinstance(x, basestring): return from_str(x, prec, rounding)
  File "/usr/lib/python2.7/site-packages/mpmath/libmp/libmpf.py", line 1299, in from_str
    return from_rational(int(p), int(q), prec, rnd)
ValueError: invalid literal for int() with base 10: '-2163048125L'

== tests finished: 6841 passed, 185 skipped, 1 exceptions, in 5187.94 seconds ==
DO *NOT* COMMIT!
============================= test process starts ==============================
executable:         /usr/bin/python2  (2.7.13-final-0) [CPython]
architecture:       32-bit
cache:              yes
ground types:       gmpy 2.0.8
hash randomization: on (PYTHONHASHSEED=1855423098)
```

The given int is indeed to large to fit in 32-bit, but I guess, that should have been hanlded somewhere else, right?
Let me know if you need any more details.

cheers,
deep-42-thought

**Comment by jksuom** [comment]
This looks like an mpmath issue. It should probably use Python 2 long instead of int for unlimited precision.

**Comment by deep-42-thought** [comment]
ok, I opened an issue for mpmath, too: https://github.com/fredrik-johansson/mpmath/issues/368

**Comment by ylemkimon** [comment]
Also on Windows Python 2.7 32/64-bit.

**Comment by ylemkimon** [comment]
PRed a fix https://github.com/fredrik-johansson/mpmath/pull/374.

**Comment by ylemkimon** [comment]
A fix fredrik-johansson/mpmath#374 has been PRed and merged in the upstream. This issue will be fixed and in the next mpmath release, and I'll close this issue then. Thank you!

**Comment by oscarbenjamin** [comment]
Looks like this is fixed upstream so I'll close this.
